aptos-move/aptos-release-builder/src/components/mod.rs

```diff
@@ -29,53 +30,115 @@ pub struct ReleaseConfig {
     pub feature_flags: Option<Features>,
     #[serde(default, skip_serializing_if = "Option::is_none")]
     pub consensus_config: Option<OnChainConsensusConfig>,
+    #[serde(default)]
+    pub is_multi_step: bool,
 }
 
 impl ReleaseConfig {
     pub fn generate_release_proposal_scripts(&self, base_path: &Path) -> Result<()> {
-        let mut result = vec![];
-
-        // First create framework releases
-        if self.framework_release {
-            result.append(&mut framework::generate_upgrade_proposals(self.testnet)?);
-        }
-
-        if let Some(gas_schedule) = &self.gas_schedule {
-            result.append(&mut gas::generate_gas_upgrade_proposal(
-                gas_schedule,
-                self.testnet,
-            )?);
-        }
-
-        if let Some(version) = &self.version {
-            result.append(&mut version::generate_version_upgrade_proposal(
-                version,
-                self.testnet,
-            )?);
-        }
-
-        if let Some(feature_flags) = &self.feature_flags {
-            result.append(&mut feature_flags::generate_feature_upgrade_proposal(
-                feature_flags,
-                self.testnet,
-            )?);
-        }
-
-        if let Some(consensus_config) = &self.consensus_config {
-            result.append(&mut consensus_config::generate_consensus_upgrade_proposal(
-                consensus_config,
-                self.testnet,
-            )?);
-        }
-
-        for (idx, (script_name, script)) in result.into_iter().enumerate() {
-            let mut script_path = base_path.to_path_buf();
-            let proposal_name = format!("{}-{}", idx, script_name);
-            script_path.push(&proposal_name);
-            script_path.set_extension("move");
-
-            std::fs::write(script_path.as_path(), script.as_bytes())
-                .map_err(|err| anyhow!("Failed to write to file: {:?}", err))?;
+        let mut result: Vec<(String, String)> = vec![];
+
+        if self.is_multi_step {
```

Comment:
Is there a way we can structure the code here to avoid so much redundancy between the if else branches?

Reply:
yeah will think about it and update the pr!